test/livebook/web_socket/server_test.exs

```diff
@@ -0,0 +1,43 @@
+defmodule Livebook.WebSocket.ServerTest do
+  use Livebook.EnterpriseIntegrationCase, async: true
+
+  alias Livebook.WebSocket.Server
+
+  describe "connect/2" do
+    test "successfully authenticates the websocket connection", %{url: url, token: token} do
+      headers = [{"X-Auth-Token", token}]
+
+      assert {:ok, pid} = Server.start_link()
+      assert {:ok, :connected} = Server.connect(pid, url, headers)
+      assert Server.disconnect(pid) == :ok
+    end
```

Comment:
Why do we need a connect in both client and server version?

Reply:
The `Server` is only a temporary name to the GenServer version (async) of the `Client`